Detectors/Raw/src/RawFileReaderWorkflow.cxx

```diff
@@ -210,7 +212,7 @@ class RawReaderSpecs : public o2f::Task
 
  private:
   int mLoop = 0;                                   // once last TF reached, loop while mLoop>=0
-  size_t mTFIDaccum = 0;                           // TFId accumulator (accounts for looping)
+  uint32_t mTFCounter = 0;                         // TFId accumulator (accounts for looping)
```

Comment:
it would be grat if this functionality is moved to the `input-proxy`. e.g. with parameter --loop, it would made sure tfCounter (tfID) never looped. Without the parameter, the tfCounter would be allowed to strictly increase (with random jumps to simulate the real EPN tf distribution).

Reply:
@ironMann for me the direct injection of the TFs from raw-reader to DPL is the main scenario of its use. 
If I understand correctly what you want is an option in the o2-dpl-raw-proxy to ignore the mTFCounter in the DataHeader received from the DD (or the raw-reader running with --raw-channel-config) and to inject its own counter, increasing it in steps of 1 or in random jumps? Is it because you don't want to generate the counter in the DD?
BTW, the jumpy counter, if used instead of current DPH.timeStart, would make inefficient the time-pipelining mechanism, one would need to first adapt it.

Reply:
> and to inject its own counter, increasing it in steps of 1 or in random jumps? Is it because you don't want to generate the counter in the DD?
> BTW, the jumpy counter, if used instead of current DPH.timeStart, would make inefficient the time-pipelining mechanism, one would need to first adapt it.

Precisely this. I understand workarounds, but they should be isolated and seld contained. In this case, I should not be responsible to make the pipeline feature possible by adhering to some unreallistic tfID rules. Instead, the "faking" of tfid should be moved to DPL, where the future imrovements of the routing will not require us to update dd or raw-reader.

Reply:
The mTFCounter is what it is: counter for the injected TF, I don't see why you are calling its incrementing a "faking", it is not related to replay. What would you inject on DD level? 
The case of hacking the counter to emulate randomness of its increment on the EPN level is legitimate, I can add it as an option to o2-dpl-raw-proxy (and raw-reader) but since this render the time-pipelining inefficient, @ktf should tell when we will be ready for it.

Reply:
I might be just confused then by the comments "accounts for looping" and "unless the TF is replayed"...
What you call here tfCounter (readout calls it tfID) is a field as unique to a TF as the firstOrbit. I'm saying that the inputs (DD and reader) should not change it for the sake of making the pipeline feature working (looping or no looping). If the strict incremental increase is required at this time, it should be enforced by the DPL, and fixed later by the DPL.

Reply:
> What you call here tfCounter (readout calls it tfID) is a field as unique to a TF as the firstOrbit. 

What would be the reason for such interpretation of this field? The firstOrbit is already unique (within the run) and logically tied to TFs payload. It only "deficiency" is that in case of replay it does not allow to disentangle one TF from another, and a simple injection counter would allow to fix this. What other role would you assign to it?
My comment "accounts for looping" meant to emphasize that it is incremented also for replayed TFs.

Reply:
> What would be the reason for such interpretation of this field? What other role would you assign to it?

It is an designation created by the readout and used for data distribution, and posibly for the CTF indexing. It is completely feasible that the TFID could be used in many places: e.g. to mark bad data or bad run conditions, used in the logbook somehow, and used to report the non-successful TF distribution etc. Also, not all FLPs will have the "firstOrbit" in the data. Since the data on all FLPs will pass through some kind of DPL workflow (if nothing else, for QC) it is important that DPL leaves this field unmodified.

Reply:
If the readout provides an absolute TF ID then this changes a picture. But does it? 
As far as I know, it just provides an info to identify the change of TF, but it does not assign any ID.
Could you provide some reference?